jujugui/static/gui/src/app/store/env/go.js

```diff
@@ -1396,30 +1396,47 @@ YUI.add('juju-env-go', function(Y) {
        Set a service's charm.
 
        @method setCharm
-       @param {String} service_name The name of the service to be upgraded.
-       @param {String} charm_url The URL of the charm.
-       @param {Boolean} force Force upgrading machines in error.
+       @param {String} serviceName The name of the service to be upgraded.
+       @param {String} charmUrl The URL of the charm.
+       @param {Boolean} forceUnits Force the units when upgrading.
+       @param {Boolean} forceSeries Force the series when upgrading.
        @param {Function} callback A callable that must be called once the
          operation is performed.
        @return {undefined} Sends a message to the server only.
      */
-    setCharm: function(service_name, charm_url, force, callback) {
+    setCharm: function(serviceName, charmUrl, forceUnits, forceSeries, cb) {
       var intermediateCallback = null;
-      if (callback) {
+      if (cb) {
         intermediateCallback = Y.bind(this.handleSetCharm, this,
-            callback, service_name, charm_url);
+            cb, serviceName, charmUrl);
       }
-      this._send_rpc(
-          { Type: 'Client',
-            Request: 'ServiceSetCharm',
-            Params: {
-              ServiceName: service_name,
-              CharmUrl: charm_url,
-              Force: force
-            }
-          },
-          intermediateCallback
-      );
+      var rpc = {
+        Type: 'Service',
+        Request: 'SetCharm',
+        Params: {
+          ServiceName: serviceName,
+          CharmUrl: charmUrl,
+          ForceUnits: forceUnits,
+          ForceSeries: forceSeries
+        }
+      };
+      if (this.findFacadeVersion('Service') === null) {
```

Comment:
Are we sure this was enabled in all service versions?
I'd use the usual version check here (=== null || < 3).